nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs

```diff
@@ -0,0 +1,283 @@
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+using NuGetUpdater.Core.Analyze;
+using NuGetUpdater.Core.Discover;
+using NuGetUpdater.Core.Run.ApiModel;
+
+namespace NuGetUpdater.Core.Run;
+
+public class RunWorker
+{
+    private readonly IApiHandler _apiHandler;
+    private readonly Logger _logger;
+
+    internal static readonly JsonSerializerOptions SerializerOptions = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicy.KebabCaseLower,
+        WriteIndented = true,
+        Converters = { new JsonStringEnumConverter() },
+    };
+
+    public RunWorker(IApiHandler apiHandler, Logger logger)
+    {
+        _apiHandler = apiHandler;
+        _logger = logger;
+    }
+
+    public async Task RunAsync(FileInfo jobFilePath, DirectoryInfo repoContentsPath, string baseCommitSha, FileInfo outputFilePath)
```

Comment:
The separate runners, `DiscoverWorker`, `AnalyzeWorker`, and `UpdateWorker` each do the error checking.  Eventually, once we're fully on this runner, I'll lift the error checking here, but for now we need it in the other runners, because those are the various entry points from the current Ruby code.  I've added `// TODO:` comments to surface the errors from those workers.  An example is in this file, lines 64, 126, and 156.